File: Data Structures & Algorithms/buy-and-sell-crypto/submission-0.cs
public class Solution {
    public int MaxProfit(int[] prices) {
       int l = 0;
       int r = 1;
       int maxProfit = 0;
       int arrayLength = prices.Length;
       while(r < arrayLength)
       {
         if(prices[l] < prices[r])
         {
            int profit = prices[r]-prices[l];
            maxProfit = Math.Max(maxProfit, profit); 
         }
         else
         {
            l=r;
         }
         r++;
       }
       return maxProfit;
    }
}

File: Data Structures & Algorithms/eating-bananas/submission-0.py
import math
class Solution:
    def minEatingSpeed(self, piles: List[int], h: int) -> int:
        left = 1

        right = max(piles)
        result = 0
        while left < right:
            middle = (left + right)//2
        
            ceil_sum = 0
            for num in piles:
                ceil_sum = ceil_sum + math.ceil(num/middle)
                
            if ceil_sum <= h:
                right = middle
            else:
                left = middle+1

        return left
    
        
    
    

        
